pkg/kv/kvclient/rangefeed/BUILD.bazel

```diff
@@ -27,28 +27,43 @@ go_library(
     ],
 )
 
-# keep
+# We create a stand-alone go_library target that also includes the generated
+# mocks. Generating mocks depends on the base library being mocked (making
+# this round-about necessary). See top-level BUILD.bazel for the necessary
+# resolve directive to help other packages distinguish between the two.
+#
+# NB: We could alternatively avoid the resolve directive and have the
+# mocks-included target be named as "rangefeed" instead. That unfortunately
+# doesn't play well with gazelle, which will overwrite the "rangefeed" named
+# target whenever files/dependencies are changed.
+
+gomock(
+    name = "mock_rangefeed",
+    out = "mocks_generated.go",
```

Comment:
I think we could end up needing multiple mock files -- in particular when using source mode, since I think `mockgen` then generates one output file per input file. With [Mockery](https://github.com/vektra/mockery), it places mocks in a `mocks/` subdirectory with `package mocks`. Perhaps a scheme like that could help with dependency handling as well?

Reply:
The scheme here is generalizable to multiple mock files, we'd create a `gomock` target for each one and make sure they're listed as deps in the package's `:with-mocks` target. And yea, a separate mocks directory would help, and (test) packages that want to import mocks for another package, could import it directly. I think I'll avoid doing that for now and stick to the standard way of placing golang/mocks. We can always re-visit if this proves to be too cumbersome/decide to go with mockery.

Reply:
Yeah, I just find it a bit unfortunate that whenever someone wants to add a mock to a new package they'll not only have to muck around with the package build dependencies (which is fine enough), but also have to change the package target name, update dozens of unrelated build files, and fiddle with resolve overrides and such in the root build manifest.

It'd be nice if this was a bit less involved, and I'm thinking with separate mocks packages we might be able to get away with adding a new mock target that Gazelle could automatically depend on for any tests that need the mocks. I'm not entirely sure we can avoid circular dependencies in that case though, with internal tests that depend on the mocks.

Reply:
The two `gazelle:resolve` (in the top-level BUILD file + package's BUILD file for this annoying noop target) + `make bazel-generate` should do all of it for us. I'm also hoping neither will be necessary if the upstream issues are resolved/have easier work-arounds. Do you feel it's still too cumbersome? Or could be documented better? 

And yes, even with a separate mocks package, we'd need some bending over to work with internal tests. Curious to see if you have other ideas we could try.

Reply:
It's workable, and definitely an improvement over what we're currently doing, but a bit hacky. Don't have any better ideas off the bat though, and it's quite possible that this is as good as it's going to get right now. Might be worth looking around for some prior art.